### DISCUSSION TITLE
Overwriting FluentNavLink callback

### DISCUSSION THREAD

**Discussion by Andras-Csanyi** [microsoft/fluentui-blazor#1791] [Q&A] [answered]
I am building an application using Fluxor and for a certain event I need to know on which item the click happened and the callback returning a certain value. The menu looks like below.

The current implementation has only `MouseEventArgs` and I cannot pass any further value.

- Users (when it is clicked on its callback returns `users`)
- Roles (when it is clicked on its callback returns `roles`)

On code level it should, in my opinion, look like this:

```
<FluentNavLink @onclick="OnClick(linkName)">Users<FluentNavLink>
<FluentNavLink @onclick="OnClick(linkName)">Users<FluentNavLink>

@code {

    public string OnClick(string linkName)
    {
        // setting up state or firing event by passing the `linkName`
    }

}
```

Is there a way to do this or I need to create my own component?

**Comment by Andras-Csanyi** [comment, marked as answer]
Ahh, Copilot helped me out.

The solution is:

```
<FluentNavLink @onclick="@(e => OnClick(e, linkName))">Users<FluentNavLink>

@code {

    public string OnClick(MouseEventArgs args, string linkName)
    {
        // setting up state or firing event by passing the `linkName`
    }

}
```

**Reply by vnbaaij** [reply]
Clever!
